src/vipersci/carto/bounds.py

```diff
@@ -0,0 +1,80 @@
+"""
+The bounds module is used to calculate a bounding box for a set of 2d spatial data.
+"""
+
+# Copyright 2023, United States Government as represented by the
+# Administrator of the National Aeronautics and Space Administration.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Reuse is permitted under the terms of the license.
+# The AUTHORS file and the LICENSE file are at the
+# top level of this library.
+
+import math
+
+import numpy as np
+from numpy.typing import NDArray
+from rasterio.coords import BoundingBox
+
+
+def compute_bounds(
+    x_arr: NDArray,
+    y_arr: NDArray,
+) -> BoundingBox:
+    """Compute cartesian bounds of input coordinates
+
+    Args:
+        x_arr (NDArray): set of x locations
+        y_arr (NDArray): set of y locations
+
+    Returns:
+        BoundingBox: Rasterio bounding box of input locations
+    """
+    return BoundingBox(np.amin(x_arr), np.amin(y_arr), np.amax(x_arr), np.amax(y_arr))
+
+
+def pad_grid_align_bounds(
```

Comment:
After reading through all of these changes, I think that instead of requiring a BoundingBox object (which is just a named tuple with some aliases) and returning same, that this function should take *left, bottom, right, top* as its first four arguments and return a four-tuple.  This allows people to easily pass individual variables, or to unpack a BoundingBox or a shapely.bounds-style structure to this function without requiring that you convert your data into a BoundingBox.  This makes the function easier to use.